Sources/Resource/NetworkResource.swift

```diff
@@ -8,12 +8,47 @@
 
 import Foundation
 
-public typealias ResourceParseClosure<T> = (Data) throws -> T
+public protocol NetworkResource: Resource {
```

Comment:
How do we inject the url in the `RelativeNetworkResource` ? From my understanding, it would be inject by the network stack in the resource so it can build the `URLRequest`